src/controllers/bookController.ts

```diff
@@ -25,6 +28,39 @@ class BookController {
             error_description: 'Endpoint not implemented yet.',
         });
     }
+
+    getAllBooks(req: Request, res: Response) {
+        const allBooks: Book[] = [];
+        const connection = new TediousConnection(dbConfig);
+
+        connection.on('connect', function (err: Error | undefined) {
+            if (err) {
+                res.json(JSON.stringify(err));
+                return;
+            }
+            const request = new TediousRequest(
+                'select * from bookish.dbo.book',
+                (err: Error) => {
+                    if (err) {
+                        console.log(err);
+                    } else {
+                        // and we close the connection
+                        connection.close();
+                        res.json(JSON.stringify(allBooks));
+                    }
+                },
+            );
+
+            request.on('row', function (columns) {
+                const ISBN = columns.find((col) => col.metadata.colName=='ISBN').value;
```

Comment:
I can see that you'll use this pattern a lot; I'd pull this out into a separate function!